packages/firebase_storage/android/src/main/java/io/flutter/plugins/firebase/storage/FirebaseStoragePlugin.java

```diff
@@ -285,6 +289,50 @@ private void putData(MethodCall call, Result result) {
     result.success(handle);
   }
 
+  private void listAll(MethodCall call, final Result result) {
+    String path = call.argument("path");
+    StorageReference ref = firebaseStorage.getReference().child(path);
+    final Task<ListResult> listTask = ref.listAll();
+    listTask.addOnSuccessListener(
+        new OnSuccessListener<ListResult>() {
+          @Override
+          public void onSuccess(ListResult listResult) {
+            Map<String, Object> map = new HashMap<>();
+            map.put("pageToken", listResult.getPageToken());
+            Map<String, Object> mapItems = new HashMap<>();
+            for (int i = 0; i < listResult.getItems().size(); i++) {
+              mapItems.put(
+                  listResult.getItems().get(i).getName(),
+                  buildMapStorageReference(listResult.getItems().get(i)));
+            }
+            map.put("items", mapItems);
+            Map<String, Object> mapPrefixes = new HashMap<>();
+            for (int i = 0; i < listResult.getPrefixes().size(); i++) {
+              mapItems.put(
```

Comment:
I was just incidentally reading through this PR and noticed that there appears to be a bug here, where this is calling `put` against `mapItems` instead of `mapPrefixes`.